internal/bminventory/inventory_test.go

```diff
@@ -2083,6 +2083,61 @@ var _ = Describe("cluster", func() {
+					OpenshiftVersion:     swag.String(insufficientOpenShiftVersionForNoneHA),
+					PullSecret:           swag.String("{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"),
+					HighAvailabilityMode: &noneHaMode,
+				},
+			})
+			verifyApiError(reply, http.StatusBadRequest)
+		})
+		It("create non ha cluster success, release version is grater than minimal", func() {
+			bm.clusterApi = cluster.NewManager(cluster.Config{}, common.GetTestLog().WithField("pkg", "cluster-monitor"),
+				db, mockEvents, nil, nil, nil, nil, nil, nil, nil, nil)
+			noneHaMode := models.ClusterHighAvailabilityModeNone
+			insufficientOpenShiftVersionForNoneHA := "4.8.0"
+			reply := bm.RegisterCluster(ctx, installer.RegisterClusterParams{
+				NewClusterParams: &models.ClusterCreateParams{
+					Name:                 swag.String("some-cluster-name"),
+					OpenshiftVersion:     swag.String(insufficientOpenShiftVersionForNoneHA),
+					PullSecret:           swag.String("{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"),
+					HighAvailabilityMode: &noneHaMode,
+				},
+			})
+			Expect(reflect.TypeOf(reply)).Should(Equal(reflect.TypeOf(installer.NewRegisterClusterCreated())))
+			actual := reply.(*installer.RegisterClusterCreated)
+			Expect(actual.Payload.HighAvailabilityMode).To(Equal(swag.String(noneHaMode)))
+			Expect(actual.Payload.UserManagedNetworking).To(Equal(swag.Bool(true)))
+			// verify VipDhcpAllocation was set to false even though it was sent as true
+			Expect(actual.Payload.VipDhcpAllocation).To(Equal(swag.Bool(false)))
+		})
+		It("create non ha cluster success, release version is pre-release and grater than minimal", func() {
+			bm.clusterApi = cluster.NewManager(cluster.Config{}, common.GetTestLog().WithField("pkg", "cluster-monitor"),
+				db, mockEvents, nil, nil, nil, nil, nil, nil, nil, nil)
+			noneHaMode := models.ClusterHighAvailabilityModeNone
+			insufficientOpenShiftVersionForNoneHA := "4.8.0-fc.2"
+			reply := bm.RegisterCluster(ctx, installer.RegisterClusterParams{
+				NewClusterParams: &models.ClusterCreateParams{
+					Name:                 swag.String("some-cluster-name"),
+					OpenshiftVersion:     swag.String(insufficientOpenShiftVersionForNoneHA),
+					PullSecret:           swag.String("{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"),
+					HighAvailabilityMode: &noneHaMode,
+				},
+			})
+			Expect(reflect.TypeOf(reply)).Should(Equal(reflect.TypeOf(installer.NewRegisterClusterCreated())))
+			actual := reply.(*installer.RegisterClusterCreated)
+			Expect(actual.Payload.HighAvailabilityMode).To(Equal(swag.String(noneHaMode)))
+			Expect(actual.Payload.UserManagedNetworking).To(Equal(swag.Bool(true)))
+			// verify VipDhcpAllocation was set to false even though it was sent as true
+			Expect(actual.Payload.VipDhcpAllocation).To(Equal(swag.Bool(false)))
+		})
```

Comment:
nit: maybe also a test for release version equals minimal

Reply:
That is the first tc (happy flow)